bin/test.bash

```diff
@@ -41,6 +41,43 @@ function usage {
   exit 255
 }
 
+# Define cleanup() function. This needs to be defined ASAP to catch SIGINT and SIGTERM at the earliest signs of trouble.
+cleanup() {
+  echo "Cleaning up..."
+
+  # Terminate the dev server if it's running
+  if [ -n "$DEV_GID" ]; then
+    echo "Terminating the dev server..."
+    kill -INT -- -"$DEV_GID"
+    echo "Dev server terminated."
+  fi
+
+  # Terminate the builder if it's running
+  if [ -n "$BUILDER_GID" ]; then
+    echo "Terminating the builder..."
+    kill -INT -- -"$BUILDER_GID"
+    echo "Builder terminated."
+  fi
+
+  # Terminate the preview server if it's running
+  if [ -n "$PREVIEW_GID" ]; then
+    echo "Terminating preview server..."
+    kill -INT -- -"$PREVIEW_GID"
+    echo "Preview server terminated."
+  fi
+
+  # Terminate the live builder if it's running
+  if [ -n "$LIVE_GID" ]; then
+    echo "Terminating the builder..."
+    kill -INT -- -"$LIVE_GID"
+    echo "Builder terminated."
+  fi
+  # Exit the script
+  exit "$EXIT_CODE"
```

Comment:
The $EXIT_CODE is not set here when `cleanup` is called by `trap`.  So I'm seeing this error when I use `Ctrl-C` to terminate `test.bash`:

![image](https://github.com/FarmData2/FarmData2/assets/16962482/b0482821-3c1a-4127-b173-d78f210d4811)

Maybe pass the `$EXIT_CODE` in as a parameter from line 343 and use a default of `130` (Terminated by `Ctrl-C`) if `cleanup` is called by the `trap`?

Reply:
> The $EXIT_CODE is not set here when `cleanup` is called by `trap`. So I'm seeing this error when I use `Ctrl-C` to terminate `test.bash`:
> 
> ![image](https://private-user-images.githubusercontent.com/16962482/299346096-b0482821-3c1a-4127-b173-d78f210d4811.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxMDc5MzQsIm5iZiI6MTcwNjEwNzYzNCwicGF0aCI6Ii8xNjk2MjQ4Mi8yOTkzNDYwOTYtYjA0ODI4MjEtM2MxYS00MTI3LWIxNzMtZDc4ZjIxMGQ0ODExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI0VDE0NDcxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMjhiMmI4MmI5ZTdmMGNmNzM4NDhmOTcxODVmMzRjMWQ5ZTU2MGI3MzAzOWViMThkZmE2YTkxMWRjODAwNDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NRsIG8-WVGd6tHfp80dusuxX17K9NBQgpcaDGNIp9JA)
> 
> Maybe pass the `$EXIT_CODE` in as a parameter from line 343 and use a default of `130` (Terminated by `Ctrl-C`) if `cleanup` is called by the `trap`?

I was worried about that. I was considering using the standard 0 but 130 should be fine. I wasn't sure what exit code to use, I should've had the foresight to realize that. Let me change it 130 real quick.

Reply:
Exit Code changed.